clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -3335,8 +3335,20 @@ void prepareTypeConverter(mlir::LLVMTypeConverter &converter,
                           mlir::DataLayout &dataLayout) {
   converter.addConversion([&](mlir::cir::PointerType type) -> mlir::Type {
     // Drop pointee type since LLVM dialect only allows opaque pointers.
+
+    auto addrSpace =
+        mlir::cast_if_present<mlir::cir::AddressSpaceAttr>(type.getAddrSpace());
+    // null addrspace attribute indicates the default addrspace
+    if (!addrSpace)
+      return mlir::LLVM::LLVMPointerType::get(type.getContext());
+
+    // TODO(cir): Query the target-specific address space map to lower other ASs
+    // like `opencl_private`.
+    assert(!MissingFeatures::targetLoweringInfoAddressSpaceMap());
+    assert(addrSpace.isTarget() && "NYI");
```

Comment:
Assert/unrecheable is usually in terms of CIRGen, here a `return op.emitError...` is preferred.

Reply:
Here in the context of TypeConverter, do we have a proper way to emit IR error? Should I use `llvm_unreachable` instead?

Reply:
You'd have to check the generated code to see if there's a better way, but `llvm_unreachable` sounds good to me too.